bioconda_utils/build.py

```diff
@@ -0,0 +1,351 @@
+import subprocess as sp
+from collections import defaultdict
+import os
+import logging
+import networkx as nx
+from . import utils
+from . import docker_utils
+from . import pkg_test
+from conda_build import api
+
+logger = logging.getLogger(__name__)
+
+
+def build(recipe,
+          recipe_folder,
+          env,
+          testonly=False,
+          mulled_test=True,
+          force=False,
+          channels=None,
+          docker_builder=None):
+    """
+    Build a single recipe for a single env
+
+    Parameters
+    ----------
+    recipe : str
+        Path to recipe
+
+    env : dict
+        Environment (typically a single yielded dictionary from EnvMatrix
+        instance)
+
+    testonly : bool
+        If True, skip building and instead run the test described in the
+        meta.yaml.
+
+    mulled_test : bool
+        Test the built package in a minimal docker container
+
+    force : bool
+        If True, the recipe will be built even if it already exists. Note that
+        typically you'd want to bump the build number rather than force
+        a build.
+
+    channels : list
+        Channels to include via the `--channel` argument to conda-build
+
+    docker_builder : docker_utils.RecipeBuilder object
+        Use this docker builder to build the recipe, copying over the built
+        recipe to the host's conda-bld directory.
+    """
+    env = dict(env)
+    logger.info(
+        "BIOCONDA BUILD START %s, env: %s",
+        recipe, ';'.join(['='.join(map(str, i)) for i in sorted(env.items())])
```

Comment:
Sorry, I should have remembered that myself, since it is my own implementation ;-). Still, I think you can get rig of the `map(str, i)` then. A join on the tuple should do.